keylime/cmd/sign_runtime_policy.py

```diff
@@ -0,0 +1,111 @@
+import argparse
+import sys
+from typing import NoReturn
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.serialization import load_pem_private_key
+
+from keylime.dsse import dsse, ecdsa, x509
+
+# pylint: disable=pointless-string-statement
+"""
+This script signs Keylime runtime policies using DSSE.
+
+Example usage:
+
+To generate an ECDSA key and sign a runtime policy:
+
+```
+python sign_runtime_policy.py -r <runtime policy path> -o keylime-policy-signed.json
+```
+
+To sign a runtime policy with an inline x509 certificate using a user-provided key:
+
+```
+python sign_runtime_policy.py -r <runtime policy path> -k private.pem -b x509 -o keylime-policy-signed.json
+```
+
+To view help and see all available options:
+
+```
+python sign_runtime_policy.py -h
+```
+"""
+
+
+def main() -> None:
+    parser = ConversionParser()
+    parser.add_argument("-r", "--runtime_policy", help="Runtime policy file location", action="store")
+    parser.add_argument("-k", "--keyfile", help="Private key to sign policy with", action="store")
+    parser.add_argument(
+        "-p",
+        "--keypath",
+        help="Path to write created private key if -k is not specified",
+        action="store",
+        default="keylime-ecdsa-key",
+    )
+    parser.add_argument("-b", "--backend", help="DSSE backend to use", action="store", default="ecdsa")
+    parser.add_argument("-o", "--output_file", help="Output file path", action="store")
+
+    # print help if no arguments provided
+    if len(sys.argv) == 1:
+        parser.print_help()
+        parser.exit()
+
+    args = parser.parse_args()
+    if not args.runtime_policy:
+        print("Runtime policy is required!")
+        sys.exit(1)
+    elif args.backend not in ["ecdsa", "x509"]:
+        print("Unsupported DSSE backend!")
+        sys.exit(1)
+    elif args.keyfile and args.keypath:
+        print("Only one of -k and -p may be specified at once!")
```

Comment:
This doesn't seem to work:
```
# /usr/local/bin/keylime_sign_runtime_policy -b ecdsa -o mypolicy-signed.json -k ecdsa-priv.pem -r mypolicy.json                                                                  Only one of -k and -p may be specified at once!
```

Reply:
Ah, thanks for the catch - because there was a default value for `--keypath`, that condition would trip if `--keyfile` was specified. It's been fixed.